app/src/main/java/de/westnordost/streetcomplete/overlays/surface/RoadSurfaceOverlay.kt

```diff
@@ -0,0 +1,115 @@
+import de.westnordost.streetcomplete.osm.surface.Surface
+import de.westnordost.streetcomplete.osm.surface.SurfaceMissing
+import de.westnordost.streetcomplete.osm.surface.SurfaceMissingWithNote
+import de.westnordost.streetcomplete.osm.surface.UNDERSPECIFED_SURFACES
+import de.westnordost.streetcomplete.osm.surface.keysToBeRemovedOnSurfaceChange
+import de.westnordost.streetcomplete.osm.surface.createSurfaceStatus
+import de.westnordost.streetcomplete.overlays.Color
+import de.westnordost.streetcomplete.overlays.Overlay
+import de.westnordost.streetcomplete.overlays.PolygonStyle
+import de.westnordost.streetcomplete.overlays.PolylineStyle
+import de.westnordost.streetcomplete.overlays.StrokeStyle
+import de.westnordost.streetcomplete.overlays.Style
+import de.westnordost.streetcomplete.quests.surface.AddPathSurface
+import de.westnordost.streetcomplete.quests.surface.AddRoadSurface
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.BICYCLIST
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.CAR
+
+class RoadSurfaceOverlay : Overlay {
+
+    override val title = R.string.overlay_road_surface
+    override val icon = R.drawable.ic_quest_street_surface
+    override val changesetComment = "Specify road surfaces"
+    override val wikiLink: String = "Key:surface"
+    override val achievements = listOf(CAR, BICYCLIST)
+    override val hidesQuestTypes = setOf(AddRoadSurface::class.simpleName!!, AddPathSurface::class.simpleName!!)
+
+    private val handledSurfaces = Surface.values().map { it.osmValue }.toSet() + INVALID_SURFACES
+
+    override fun getStyledElements(mapData: MapDataWithGeometry): Sequence<Pair<Element, Style>> {
+        return mapData.filter("""
+               ways, relations with
+                 highway ~ ${(ALL_ROADS).joinToString("|")}
+                 and (!surface or surface ~ ${handledSurfaces.joinToString("|")})
+                 and (!surface:note or surface)
+           """)
+           .filter { element -> tagsHaveOnlyAllowedSurfaceKeys(element.tags) }
+           .map { it to getStyle(it) }
+    }
+
+    // https://taginfo.openstreetmap.org/search?q=surface
+    private val supportedSurfaceKeys = listOf(
+        // supported in this overlay, but not in all overlays
+        "sidewalk:both:surface", "sidewalk:right:surface", "sidewalk:left:surface", "sidewalk:surface",
+
+        // this is not a valid tag on road and therefore not supported here
+        // "footway:surface", "cycleway:surface",
+
+        // supported in both surface overlays
+        "surface", "surface:note"
+    ) + keysToBeRemovedOnSurfaceChange("")
+
+    private val allowedTagWithSurfaceInKey = supportedSurfaceKeys + listOf(
+        "proposed:surface", // does not matter
+    )
+
+    private fun tagsHaveOnlyAllowedSurfaceKeys(tags: Map<String, String>): Boolean {
+        return tags.keys.none {
+            "surface" in it && it !in allowedTagWithSurfaceInKey
+        }
+    }
```

Comment:
i don't understand why you think this is necessary.

Why should it not be possible to edit the road's surface because someone added a tag like [`surface_survey`](https://taginfo.openstreetmap.org/keys/surface_survey), `surface:colour`, `source:surface` or whatever? From my point of view, this will only lead to bug reports of confused users that wonder why they cannot edit (or see, for that matter) the surface of certain roads. The presence of some other `.*surface.*` tag doesn't make the recording and updating of the `surface` tag proper invalid, does it?

(Same for the path overlay)

Reply:
`source:surface` is not blocking because it is in `keysToBeRemovedOnSurfaceChange`

But lets say that it is not listed there and would be not removed on edit with StreetComplete. That would result in `source:surface` remaining and containing misleading and invalid data. So to prevent that edit is blocked.

I can review more `surface` including tags to get rate of unexplainably blocked editing even lower (I would either whitelist them or add them to tags removed on editing surface value).

Reply:
I suspected that this would be the reason why you added that. But I find this argument faulty. In reality, you could make up for **any** tag any number of additional related tags that _should_ be deleted/updated/something when that "main" tag changes. Since there is no limit to the potential tags that _could_ be added in relation to any other tags due to the "any tags you like" policy, I find the endeavor to kind of take these into account quixotic.

For example, technically, you'd need to add this behavior to every single quest that updates any tag.

Reply:
> in reality, you could make up for any tag any number of additional related tags that should be deleted/updated/something when that "main" tag changes. 

Definitely, though making attempt to support popular and well formed ones (with `surface` in tag name) can be doable.

On the other hand, making honest attempt to support valid values used at least a bit may be better idea than locking editing in way that appears random to users.

So maybe removing this editing filter would be a good idea...

After all, any damage caused here will be caused also by iD users and I do not remember major complaints about that.

Reply:
Though I would keep it even if it is a bit quixotic and may require extra maintenance, but you are clearly against and I do not really disagree with your arguments.

Though I would defend removing known keys that are invalidated by surface change.